### PR TITLE
Issue #108: SuppressionPatchFilter: new property 'neverSuppressedChecks'

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #108: SuppressionPatchFilter: new property 'neverSuppressedChecks'

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java [hunk 1]
```diff
@@ -99,13 +110,34 @@ private boolean isLineMatching(AuditEvent event) {
...
+     * Is matching by never suppress check.
+     *
+     * @param event event
+     * @return true if it is matching
+     */
+    private boolean isNeverSuppressCheck(AuditEvent event) {
+        boolean result = false;
+        final String checkShortName = getCheckShortName(event);
+        if (neverSuppressedChecks != null) {
+            if (neverSuppressedChecks.contains(checkShortName)) {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java:132]: Please match by `id` also, please read issue attantively

Comment (HuGanghui) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java:142]: done

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java:142]: I meant to use single property that keeps names and ids. From event we take name and I'd and try to match.

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchfilter/neversuppressedids/RegexpMultiline/defaultContextConfig.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+          "https://checkstyle.org/dtds/configuration_1_3.dtd">
+<module name="Checker">
+  <module name="RegexpMultiline">
+    <property name="matchAcrossLines" value="true"/>
+    <property name="format" value="System\.out\.\n                print\("/>
+    <property name="id" value="1"/>
```
[the remark this answers is not in the document]
Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchfilter/neversuppressedids/RegexpMultiline/defaultContextConfig.xml]: done

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchfilter/RegexpMultiline/patchedline/expected.txt [hunk 3]
```diff
@@ -1,2 +1 @@
-Test.java:18: Line matches the illegal pattern 'System\.out\.\n                print\('.
 Test.java:28: Line matches the illegal pattern 'System\.out\.\n                print\('.
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchfilter/RegexpMultiline/patchedline/expected.txt:1]: it is not clear why 18 is removed, 18th is changed line, 28th is newline.

Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchfilter/RegexpMultiline/patchedline/expected.txt:1]: ok, violation in line before .... all good.

**Pull request merged**